### ISSUE TITLE
low output by processing shotgun metagenomic data

### ISSUE THREAD

**Issue by nduan1** [biocore/sortmerna#271]
Hi,

I was using the command shown below to run a huge dataset (forward and reverse 21 G, respectively), and trying to extract bacterial information, but only a very small amount of bacterial information can be found from the results.

Did I do something wrong on this command? Should I loosen the evalue? Will that make sense then?
Please let me know if you have any suggestions and thank you so much!

sortmerna -ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta \
-ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta \
-reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_1.fastq \
-reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_2.fastq \
-workdir /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/output \
-fastx --other --paired_in \
-blast 1 cigar qcov \
-e 10 \
-num_alignments 1 -v \
--task 4

(base) -bash-4.2$ ls -hl
total 42G
-rw-r--r--. 1 nduan1 tug2312  29M Nov 29 04:12 aligned.blast
-rw-r--r--. 1 nduan1 tug2312 154M Nov 29 03:35 aligned.fastq
-rw-r--r--. 1 nduan1 tug2312 2.2K Nov 29 01:07 aligned.log
-rw-r--r--. 1 nduan1 tug2312  42G Nov 29 03:35 other.fastq

Here is the aligned.log file
 Command:
    sortmerna -ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta -ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta -reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_1.fastq -reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_2.fastq -workdir /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/output -fastx --other --paired_in -blast 1 cigar qcov -e 1 -num_alignments 1 -v --task 4

 Process pid = 59642

 Parameters summary:
    Reference file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta
        Seed length = 18
        Pass 1 = 18, Pass 2 = 9, Pass 3 = 3
        Gumbel lambda = 0.602725
        Gumbel K = 0.329559
        Minimal SW score based on E-value = 64
    Reference file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta
        Seed length = 18
        Pass 1 = 18, Pass 2 = 9, Pass 3 = 3
        Gumbel lambda = 0.602436
        Gumbel K = 0.335011
        Minimal SW score based on E-value = 64
    Number of seeds = 2
                                                                                                                      1,3           Top

**Comment by biocodz** [comment]
`-blast 1 cigar qcov` should be `-blast "1 cigar qcov"` for once.
Could  you provide the [execution-trace](https://github.com/biocore/sortmerna#execution-trace)?

**Comment by nduan1** [comment]
[eval10_sortmerna.sge.txt](https://github.com/biocore/sortmerna/files/5630644/eval10_sortmerna.sge.txt)
[eval1_sortmerna.sge.txt](https://github.com/biocore/sortmerna/files/5630645/eval1_sortmerna.sge.txt)
Hi @biocodz ,

I attached these two trace files, the one I ran with e-value 10, the other is e-value 1, thank you!

Best,

Ning

**Comment by biocodz** [comment]
don't see anything wrong so far in the traces.
Are there actual results in your 'aligned.log' like
```
 Results:
    Total reads passing E-value threshold = 5944 (59.44)
    Total reads failing E-value threshold = 4056 (40.56)
    Total reads passing %%id and %%coverage thresholds = 2813 (28.13)
    Minimum read length = 100
    Maximum read length = 100
    Mean read length    = 100

 Coverage by database:
    C:/Users/ak/a01_code/sortmerna/data/rRNA_databases/silva-arc-16s-id95.fasta		40.04
    C:/Users/ak/a01_code/sortmerna/data/rRNA_databases/silva-bac-16s-id90.fasta		19.39
```
?

**Comment by nduan1** [comment]
Command:
    sortmerna -ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta -ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta -reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_1.fastq -reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_2.fastq -workdir /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/output -fastx --other --paired_in -blast 1 cigar qcov -e 1 -num_alignments 1 -v --task 4

 Process pid = 59642

 Parameters summary:
    Reference file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta
        Seed length = 18
        Pass 1 = 18, Pass 2 = 9, Pass 3 = 3
        Gumbel lambda = 0.602725
        Gumbel K = 0.329559
        Minimal SW score based on E-value = 64
    Reference file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta
        Seed length = 18
        Pass 1 = 18, Pass 2 = 9, Pass 3 = 3
        Gumbel lambda = 0.602436
        Gumbel K = 0.335011
        Minimal SW score based on E-value = 64
    Number of seeds = 2
    Edges = 4
    SW match = 2
    SW mismatch = -3
    SW gap open penalty = 5
    SW gap extend penalty = 2
    SW ambiguous nucleotide = -3
    SQ tags are not output
    Number of alignment processing threads = 0
    Reads file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_1.fastq
    Reads file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_2.fastq
    Total reads = 122276172

 Results:
    Total reads passing E-value threshold = 344486 (0.28)
    Total reads failing E-value threshold = 121931686 (99.72)
    Minimum read length = 18
    Maximum read length = 151
    Mean read length    = 149

 Coverage by database:
    /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta          0.13
    /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta          0.16
##########################################################################################
Command:
    sortmerna -ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta -ref /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta -reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_1.fastq -reads /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_2.fastq -workdir /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/output -fastx --other --paired_in -blast 1 cigar qcov -e 10 -num_alignments 1 -v --task 4

 Process pid = 117401

 Parameters summary:
    Reference file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta
        Seed length = 18
        Pass 1 = 18, Pass 2 = 9, Pass 3 = 3
        Gumbel lambda = 0.602725
        Gumbel K = 0.329559
        Minimal SW score based on E-value = 61
    Reference file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta
        Seed length = 18
        Pass 1 = 18, Pass 2 = 9, Pass 3 = 3
        Gumbel lambda = 0.602436
        Gumbel K = 0.335011
        Minimal SW score based on E-value = 60
    Number of seeds = 2
    Edges = 4
    SW match = 2
    SW mismatch = -3
    SW gap open penalty = 5
    SW gap extend penalty = 2
    SW ambiguous nucleotide = -3
    SQ tags are not output
    Number of alignment processing threads = 0
    Reads file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_1.fastq
    Reads file: /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/CLEAN_READS/M107/final_pure_reads_2.fastq
    Total reads = 122276172

 Results:
    Total reads passing E-value threshold = 356846 (0.29)
    Total reads failing E-value threshold = 121919326 (99.71)
    Minimum read length = 18
    Maximum read length = 151
    Mean read length    = 149

 Coverage by database:
    /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-16s-id90.fasta          0.13
    /lustre/haven/proj/UTK0124/ningd/whole_metagenome_2020/sortmerna/database/silva-bac-23s-id98.fasta          0.16

Thanks for your reply, these two are the align.log files, I tried to use eval10 and 1, it seems no difference, most of the data can't pass the E-value threshold, could I have more suggestions? Thanks!

**Comment by ekopylova** [comment]
Hello @nduan1 ,

Have you tried taking a random selection (e.g. 100-1000) of the "non-rRNA classified" reads and run them through Blast? If you are seeing many sequences being classified to rRNA, then that would signal there is something odd happening with SortMeRNA classification. 

Are you using the [new databases](https://github.com/biocore/sortmerna/releases/tag/v4.2.0) released with v4.2.0 ? I wouldn't expect a large difference however if there should be many rRNA present.

Jenya

**Comment by nduan1** [comment]
Hi Jenya,

I tried to subset the "non-rRNA classified" reads and run them through the blast, nothing popped up. And I had checked with some researchers who have similar research areas, they told me this is a reasonable percentage output. Sortmerna running perfectly great for now!

Thank you so much for your help!

Hope you all have a great holiday!

Ning
